### PR TITLE
Fixes #21425 - host selection alert should refer to param per_page

### PR DESCRIPTION
**Pull request by amirfefer**
Multiple host selection alert only mention settings[:per_page],
though pagination has been changed, now the user can change per_page via select box

This may lead to miscount hosts amount :
hosts_selection

### CODE REVIEW

**Comment by amirfefer** [comment]
[foreman test]

**Comment by ohadlevy** [comment]
@amirfefer can you please add a test? thanks.

**Comment by amirfefer** [comment]
@ohadlevy  thanks for your review.
I've added some tests

**Comment by amirfefer** [comment]
[test upgrade]

**Comment by amirfefer** [comment]
[test katello]

**Comment by amirfefer** [comment]
[test katello]

**Comment by ohadlevy** [comment]
@amirfefer please rebase - FactoryGirl was renamed to FactoryBot

**Comment by amirfefer** [comment]
Thanks @ohadlevy, I've rebased.

File: app/views/hosts/_list.html.erb
Comment (tbrisker) [inline app/views/hosts/_list.html.erb]: While you're here, a couple of minor points:
1. check-title should be "Select all items on this page"
2. uncheck-title is never shown (perhaps we should just show "Toggle all items on this page"?) and also should be rephrased.
3. When clicking the tooltip isn't removed until another click.

I won't block on this, but perhaps it's time to make the checkboxes a react component? looks like a lot of the js code is quite old and over-complicated.

File: test/integration/host_js_test.rb
Comment (tbrisker) [inline test/integration/host_js_test.rb]: you can use `FactoryBot.create_list` instead to create multiple objects of the same kind at once.

File: test/integration/host_js_test.rb
Comment (tbrisker) [inline test/integration/host_js_test.rb]: This needs to be reset after every test to the original value, or you may have unexpected results in other tests.

File: test/integration/host_js_test.rb
Comment (tbrisker) [inline test/integration/host_js_test.rb:45]: if you are creating only 2 hosts how are you sure there are more then 3?  I'm assuming this works now because of the hosts fixture, but if that ever changes it will break this test in an unexpected way. Please create enough hosts for the test you need.

**Review by tbrisker — CHANGES_REQUESTED** [review]
A few minor comments inline, thanks!

File: test/integration/host_js_test.rb
Comment (amirfefer) [reply test/integration/host_js_test.rb:45]: It's not related to hosts fixtures, one host is created before each test (as you can see at line 17).

**Comment by amirfefer** [comment]
@tbrisker thanks for your review:

-  uncheck-title is now visible 
-  tooltip triggers when hovering and disappears after a click 
-  tests have a bit changed

**Review by tbrisker — CHANGES_REQUESTED** [review]
Almost there @amirfefer, the only issue still remaining is if we select hosts on more then one page, then the hover shows e.g. "40 items selected..." (which is the correct number) while the notice shows "all 20 hosts on this page...".

**Comment by amirfefer** [comment]
@tbrisker - I think we should consider making that page nonpersistent, by removing the cookie logic.
this cookie has created to handle multiple selections, (which the new `select all host` implementation solves) and  may cause confusion and hosts misselection

**Comment by tbrisker** [comment]
@amirfefer I agree, do you want to do that as a separate PR or part of this one? 
@rohoover what do you think? currently we allow selection on multiple pages, which was introduced before we added the option to select all pages, so that people could choose more then $per_page hosts for one action. with the select all pages option, i think that could lead to unexpected results with selecting on one page, then changing to the next page, selecting some other hosts and expecting the bulk action to only apply to the hosts on the current page.

**Comment by amirfefer** [comment]
@tbrisker, I prefer to separate this to another PR.

**Comment by Rohoover** [comment]
I'm a little confused on the problem. Is it that the host counter is reporting the wrong number because of two conflicting implementations of multi-select?

**Comment by amirfefer** [comment]
@tbrisker what do you say? should we merge this, and remove cookie implementation in another PR?

File: test/integration/host_js_test.rb
Comment (tbrisker) [inline test/integration/host_js_test.rb]: Please save the previous value in the setup and restore it in the teardown, so that it is clear what happens and it isn't dependent on a fixture

**Comment by waldenraines** [comment]
@amirfefer ping

**Comment by amirfefer** [comment]
Rebased.
@tbrisker could we merge this ?

**Comment by tstrachota** [comment]
[test katello]

**Comment by tstrachota** [comment]
[test foreman]

**Comment by tstrachota** [comment]
I tested the PR and apart from @tbrisker's comment about wrong count when there are some hosts selected on another page it works fine. I prefer fixing it in a separate PR.

To avoid users' confusion we could display the notice even when there's just one host selected and keep it displayed when the pages are switched.

@tbrisker if the tests run fine, are there any other blockers that would prevent from merging this?

**Review by tbrisker — APPROVED** [review]
LGTM, will merge once tests are green.

**Comment by tstrachota** [comment]
Hmm, there are some katello failures. Not sure if related.

**Comment by tbrisker** [comment]
@amirfefer can you please rebase on latest develop?

**Comment by amirfefer** [comment]
@tbrisker 
rebased :)

**Comment by tbrisker** [comment]
Merged, thanks @amirfefer !

**Pull request merged**
